[nerdyboy98] iP

nerdyboy98 opened the pull request:
# **PandoraBot**
>_The original DukeBot, with a dash of Pepe & Peepo fun~_

**With PandoraBot, you can expect:**
- [X] Meme-fied display images
- [X] Rude replies (no hard feelings of course)
- [ ] Different Peepo reactions for different inputs/errors

**Okay but for real, these are the features:**
- Logging of tasks such as to-do tasks, deadlines, as well as upcoming events (with date displayed)
- `list` function to display all the task you have input
- `sort` function to sort all available tasks in alphabetical order
- `find` function to search if a tasks exist (in case you forgot that you have already logged them in)
- `help` function in case you ~~are dumb~~ do not know the input syntaxes

Do feedback if you identify any bugs, free cookie will be given! <sub>jk</sub>

Commit:
Add Gradle support

nerdyboy98 committed:
Level 1. Greet, Echo, Exit

nerdyboy98 committed:
Level 2. Add, List

nerdyboy98 committed:
Level 3. Mark as Done

nerdyboy98 committed:
Level 4. ToDos, Events, Deadlines

nerdyboy98 committed:
Level 5. Handle Errors

nerdyboy98 committed:
Level 6. Delete

nerdyboy98 committed:
TextUiTesting

nerdyboy98 committed:
Add file loading and saving

nerdyboy98 committed:
Add Date Time recognition

nerdyboy98 committed:
Merge branch 'branch-Level-7'

nerdyboy98 committed:
Merge branch 'branch-Level-8'

nerdyboy98 committed:
Add Parser, TaskList , Ui classes

nerdyboy98 committed:
Add packages to classes

nerdyboy98 committed:
Merge branch 'add-gradle-support'

anchengyang commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,36 @@
+        this.isDone = false;
+    }
+
+    public String toggleStatus() {
```
Maybe toggleStatus could be made private (if it isnt referenced in other classes)?

anchengyang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+
```
Maybe this empty line could be removed?

anchengyang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+        this.fileManager = fileManager;
+        this.taskList = taskList;
+    }
+    public void parse(String userMessage) {
```
This method might be abit too long. Maybe you can abstract each of the code in each block? [Avoid long methods](https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html)

anchengyang reviewed (commented):
Well done so far. Just a few code quality comments 😄

nerdyboy98 committed:
Add TaskListTest.java

nerdyboy98 committed:
Change archiveBaseName and mainClassName in build.gradle

nerdyboy98 committed:
Add JavaDoc to all classes

nerdyboy98 committed:
Add find function

nerdyboy98 committed:
Changed layout to comply with coding standards

nerdyboy98 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/task/Task.java

nerdyboy98 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/FileManager.java
#	src/main/java/duke/task/TaskList.java

KevinEyo1 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,22 @@
+package duke;
+
+public class Ui {
```
Perhaps you could encapsulate more of the Ui related details within the Ui class?

KevinEyo1 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+     * <p>
+     * @return String in the format: [taskType][status] description /by DD-MMM-YYYY
+     */
+    public String toString() {
+        String formattedByDate = this.by.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+        return "[" + this.taskType + "][" + super.status + "] "
+                + super.description
+                + " (BY: " + formattedByDate
+                + ")";
+    }
```
Perhaps you could have a toString method within your Task class and override it?

KevinEyo1 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,51 @@
+public class Task implements Serializable {
+    protected String status;
+    protected String description;
+    protected boolean isDone;
```
I like that your boolean is intuitive and named like a question.

KevinEyo1 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+
+    public void add(Task newTask) {
+        this.taskList.add(newTask);
+        System.out.println("    Got it. I've added this task:");
```
Could this be extracted out to your Ui class, considering it is dealing with printing out text?

KevinEyo1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,131 @@
+            }
+        } else if (parts[0].equals("deadline")) {
+            try {
+                String[] deadlineParts = parts[1].split(" /");
+                String[] deadline = deadlineParts[1].split(" ", 2);
```
Perhaps this code can be encapsulated within your Parser class?

KevinEyo1 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,106 @@
+    public void remove(int taskNumber) {
+        try {
+            int taskIndex = taskNumber - 1;
+            Task taskToRemove = this.taskList.get(taskIndex);
```
I like the intuitive naming and appropriate camel case.

KevinEyo1 reviewed (commented):
LGTM, just some nits to fix.

nerdyboy98 committed:
Add checkstyle.xml and suppressions.xml

nerdyboy98 committed:
Add GUI to Duke

nerdyboy98 committed:
Add assertion to mark, unmark, delete, find.
Refactored Parser.parse()

nerdyboy98 committed:
Remove magic numbers and magic literals

nerdyboy98 committed:
Merge pull request #2 from nerdyboy98/branch-A-CodeQuality

Remove magic numbers and magic literals

nerdyboy98 committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/Parser.java

nerdyboy98 committed:
Merge pull request #3 from nerdyboy98/branch-A-Assertions

Add assertion to mark, unmark, delete, find.

nerdyboy98 committed:
Add SortAlphabetical.java and sort function

Bot does not have a sort task function.

Sort function allow tasks to be sorted in alphabetical order for easy
eyeballing.

Adding sort method in TaskList class and additional else if statement in
Parser class to take in "sort" command from user. Comparator class
SortAlphabetical allows for alphabetical sorting in the sort method.

Using comparator class reduces the additional lines of codes required
in TaskList class to manually compare each task description.

nerdyboy98 committed:
Update README.md

README.md does not contain instructions that tells the user how to use
the bot.

Update README.md with instructions in GFMD format on what each command
does and what format should the use input for the commands to work.

Instructions written in GFMD format for better visualization and
formatting

nerdyboy98 committed:
Change Ui image name

nerdyboy98 committed:
Add citation in Main.java, MainWindow.java, DialogBox.java
& Launcher.java

nerdyboy98 committed:
Add colours for textbox and button
Add relative file path for FileManager.java
